hack/builder/Dockerfile

```diff
@@ -136,4 +136,8 @@ RUN curl -L -o /usr/bin/bazel https://github.com/bazelbuild/bazel/releases/downl
 # Add /root/go/src/kubevirt.io/kubevirt and mark it as a safe directory for git
 RUN mkdir -p /root/go/src/kubevirt.io/kubevirt && git config --global --add safe.directory /root/go/src/kubevirt.io/kubevirt
 
+RUN [ "$(arch)" == "s390x" ] \
+     && ln /usr/bin/gcc /usr/bin/s390x-linux-gnu-gcc \
+     || echo "Not running on s390x, skip linking gcc binary"
+
```

Comment:
What is this code supposed to do?

`s390x-linux-gnu-gcc` is the name of the cross-compiler, so if anything you'd want it when **not** running on s390x.

But code in the Dockerfile runs at the time when the image is built, so that doesn't feel like the appropriate place to make this kind of decision anyway...

Reply:
I agree, but for some reason golang expects gcc to be `/usr/bin/s390x-linux-gnu-gcc` on s390x. The binary does even exist, but on centos/rhel it is named `s390x-redhat-linux-gcc`...
Hence this fix. But i guess this change is dependent on the discussion on how to build kubevirt for s390x anyway.

Reply:
For now i dropped it, since without bazel building the builder on s390x requires other changes to the Dockerfile anyway and it is not a priority at the moment.